### DISCUSSION TITLE
Adaptierung config.json auch für VeDirect ports

### DISCUSSION THREAD

**Discussion by khasselt** [hoylabs/OpenDTU-OnBattery#90] [Ideas]
Hi Helge,

ist es möglich die VeDirect pins auch in die pinconfig.json mit aufzunehmen ? 
Bzw das Codemäßig zu aktiveren dass diese auch gesteuert werden können?
Das übersteigt leider meine DEV Kenntnisse um einiges

Großes Lob für dieses Projekt

Grüße

**Comment by berni2288** [comment]
Du kannst die Pins im platformio_override.ini überschreiben.

**Reply by khasselt** [reply]
Ja ich weiß, aber es gibt dafür ja ein Feature :) mit der pinconfig.json. 
Es gibt in Github ja keinen Tickettyp "Featurerequest" sondern nur Discussion und Issue :)

**Comment by berni2288** [comment]
Ah verstehe :) Du willst es über die UI konfigurieren können.

**Reply by khasselt** [reply]
Ja konfigurieren oder auch mal überschreiben. Aktuell ist ja sonst das Profil Nutzlos, weil ich damit ja meine config überschreibe (aus der ini)

**Comment by helgeerbe** [comment]
Ich schaue mir das mal an. Das Feature kam erst spät dazu, so dass ich es mir bisher nicht angeschaut habe. Aber wenn die anderen Pins darüber konfiguriert werden können, sollte das schnell eingebaut sein.

**Reply by khasselt** [reply]
:)
Nice
Hab das ganze auch mal im Discord angesprochen.
Noby hat dazu auch schon seinen Senf gegeben. 
--> https://discord.com/channels/984173303147155506/1070072107653935205

**Comment by helgeerbe** [comment]
Ich habe den DeviceManager angepasst. Die Victron TX/RX PINS werden jetzt angezeigt. Wenn ich alle nötigen Stellen gefunden habe, müsste auch das Profil geladen werden können. Das habe ich aber noch nicht versucht, da ich aktuell nur eine produktive OpenDTU Victron habe. Mutige also vor.

Das Json File müsste ungefähr so aussehen:

````json
[
    {
        "name": "Generic NodeMCU 38 pin",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": -1,
            "clk_mode": -1
        },
        "victron": {
            "rx": 22,
            "tx": 21
        }
    }
]

**Comment by helgeerbe** [comment]
Habe doch mal das File hier hochgeladen.

```` json
[
    {
        "name": "D1 mini esp32",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 17,
            "cs": 5
        },
        "victron": {
            "rx": 22,
            "tx": 21
        }
    }
]
````

Keine Änderung meiner Pinbelegung. Aber es wird korrekt angezeigt und OpenDTU läuft noch. Wäre jetzt spannend zu wissen, ob das Pinmapping tatsächlich funktioniert.

**Reply by khasselt** [reply]
Coole Sache. Verständnishalber : GND muss nicht definiert werden sondern nur an einen so definierten Pin vom ESP angesteckt werden ?

**Reply by helgeerbe** [reply]
JA, GND von Victron geht an GND am ESP. Nur die aktiven PINS musst du konfigurieren.

Siehe auch: https://github.com/helgeerbe/OpenDTU_VeDirect/issues/58#issuecomment-1387572331
